invenio_rdm_records/fixtures/vocabularies.py

```diff
@@ -106,84 +109,249 @@ def __iter__(self):
+        (importlib_metadata backport) for entrypoints:
+        - https://docs.python.org/3/library/importlib.metadata.html
+        - https://packaging.python.org/guides/creating-and-discovering-plugins/
+          #using-package-metadata
 
-            with open(filepath) as fp:
-                data = yaml.safe_load(fp) or {}
-                for id_, entry in data.items():
-                    if isinstance(entry["data-file"], dict):
-                        entry["data-file"] = {
-                            k: path / v for k, v in entry["data-file"].items()
-                        }
-                    else:
-                        entry["data-file"] = path / entry["data-file"]
+        but Invenio is much invested in pkg_resources (``entry_points``
+        fixture assumes it). So we use pkg_resources for now until
+        _entry_points implementation can be changed.
+        """
+        return list(
+            pkg_resources.iter_entry_points('invenio_rdm_records.fixtures')
+        )
+
+    def load(self):
+        """Load the fixtures.
+
+        Loads in priority
+
+        1- app_data_folder
+        2- extensions
+        3- this package's fixtures
+
+        Fixtures found later are ignored.
+        """
+        self._loaded_vocabularies = set()
 
-                    yield id_, entry
+        # 1- Load from app_data_folder
+        filepath = self._app_data_folder / self._filename
+        # An instance doesn't necessarily have custom vocabularies
+        # and that's ok
+        if filepath.exists():
+            self.load_vocabularies(filepath)
+
+        # 2- Load from extensions / entry_points
+        self.load_from_extensions()
+
+        # 3- Load any default fixtures from invenio_rdm_records
+        self.load_vocabularies(self._pkg_data_folder / self._filename)
+
+    def load_from_extensions(self):
+        """Load vocabularies from extensions.
+
+        There might be priority conflicts at the extensions level. An
+        exception is raised in this case rather than loading any fixture.
+        """
+        # First check if any conflicts
+        vocabulary_modules = defaultdict(list)
+        extensions = [ep.load() for ep in self._entry_points()]
+        for module in extensions:
+            directory = Path(module.__file__).parent
+            filepath = directory / self._filename
+            for v in self.peek_vocabularies(filepath):
+                vocabulary_modules[v].append(module.__name__)
+
+        errors = [
+            f"Vocabulary '{v}' cannot have multiple sources {ms}"
+            for v, ms in vocabulary_modules.items() if len(ms) > 1
+        ]
+        if errors:
+            raise ConflictingFixturesError(errors)
+
+        # Then load
+        for module in extensions:
+            directory = Path(module.__file__).parent
+            filepath = directory / self._filename
+            self.load_vocabularies(filepath)
+
+    def peek_vocabularies(self, filepath):
```

Comment:
what about something like `read_vocabularies_locations` or even `read_locations`? I also feel that every function in this class has `vocabularies` on its name and it's sort of understood from being instide a class that has the name `Vocabularies` on it... e.g. `File` has a `.read`, `.write` method not `.read_file`, `.write_file`.